src/foam/dao/LRUCachingDAO.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.dao',
+  name: 'LRUCachingDAO',
+  extends: 'foam.dao.ProxyDAO',
+  implements: [ 'foam.mlang.Expressions' ],
+
+  requires: [
+    'foam.dao.MDAO',
+    'foam.dao.QuickSink'
+  ],
+
+  documentation: `
+     Least Recently Used CachingDAO :
+     Manages a DAO\'s size by removing old items. Commonly applied inside a cache to limit the cache\'s size.
+  `,
+
+  classes: [
+    {
+      /** Links an object id to a last-accessed seqNo */
+      name: 'LRUCacheItem',
+      properties: [
+        {
+          name: 'id',
+        },
+        {
+          class: 'Int',
+          name: 'seqNo',
+          documentation: 'The lowest value is the oldest'
+        },
+        {
+          class: 'Boolean',
+          name: 'hasSourceObj',
+          value: false,
+          documentation: 'Flag for checking if there is the original object in the delegate too'
+        }
+      ]
+    }
+  ],
+
+  properties: [
+    {
+      class: 'Int',
+      name: 'maxSize',
+      value: 10000,
+      documentation: 'The maximum size to allow the target dao to be'
+    },
+    {
+      name: 'trackingDAO',
+      of: 'foam.dao.DAO',
+      factory: function() {
+        return this.MDAO.create({ of: this.LRUCacheItem });
+      },
+      documentation: 'Tracks the age of items in the target dao'
+    },
+    {
+      class: 'Int',
+      name: 'lastTimeUsed_',
```

Comment:
The name 'lastTimeUsed_' is misleading since it isn't a time. Maybe something like 'nextSeqNo' would be better.